src/sidebar/components/select-filter.js

```diff
@@ -0,0 +1,69 @@
+import { createElement } from 'preact';
+import propTypes from 'prop-types';
+
+import Menu from './menu';
+import MenuItem from './menu-item';
+import SvgIcon from '../../shared/components/svg-icon';
+
+/**
+ * @typedef {import('../store/modules/filters').FilterOption} FilterOption
+ */
+
+/**
+ * @typedef SelectFilterProps
+ * @prop {FilterOption} defaultOption
+ * @prop {string} [icon]
+ * @prop {(selectedFilter: FilterOption) => any} onSelect
+ * @prop {FilterOption[]} options
+ * @prop {FilterOption} [selectedOption]
+ * @prop {string} title
+ */
+
+/**
+ * A select-element-like control for selecting one of a defined set of
+ * options.
+ *
+ * @param {SelectFilterProps} props
+ */
+function SelectFilter({
```

Comment:
@robertknight You had asked at one point what we can do when there are too many filter options to make a select-list-style component feasible. I'm hoping that by managing UI in a component here, we can adapt, like, here. Perhaps we could set a threshold over which we convert to a text `input` with an `AutocompleteList` or similar. We could also go the _other_ way and render as radio buttons if there were few enough options (this is a common treatment for facets but requires some vertical space, which we'll have if we start laying out facets in the left of the grid on wider viewports. Yep, lots to think through still, but want to make sure we are putting the foundations in place...).

Reply:
Starting out with a menu and potentially replacing it with a different component in future sounds reasonable to me. We may want to consider unifying it with the component used to select users in the LMS grading bar in future - I haven't thought through whether that will be a good fit yet.